src/components/characterList/CharactersQuery.tsx

```diff
@@ -5,43 +5,107 @@ import { SearchBar } from "../searchBar/SearchBar";
 import { Character } from "./iCharacter";
 import { Loading } from "./Loading";
 import { Pagination } from "../pagination/Pagination";
+import { useDebounce } from "../../hooks/useDebounce";
+import Status from "../Filters/Status";
+import Gender from "../Filters/Gender";
+import Species from "../Filters/Species";
+import Button from "../UI/Button";
+import CheckBoxList from "../Filters/CheckBoxList";
 import axios from "axios";
 
 type GetCharacterResponse = {
   info: { pages: number };
   results: Array<Character>;
 };
 
-async function getCharacters(filter: string, page: number) {
-  const { data } = await axios.get<GetCharacterResponse>(
-    `https://rickandmortyapi.com/api/character/?name=${filter}&page=${page}`
-  );
-  return data;
-}
-
 const CharacterQuery = () => {
   const [filter, setFilter] = useState<string>("");
   const [page, setPage] = useState<number>(1);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  const [showStatus, setShowStatus] = useState<boolean>(false);
+  const [statusValue, setStatusValue] = useState<string>("");
+
+  const [showGender, setShowGender] = useState<boolean>(false);
+  const [genderValue, setGenderValue] = useState<string>("");
+
+  const [showSpecie, setShowSpecie] = useState<boolean>(false);
+  const [specieValue, setSpecieValue] = useState<string>("");
```

Comment:
When working with typescript, it's good to use the inferrence of the language as much as possible. In all of these useState hooks you can 100% remove all of the types you're specifying because they are literally the same as the inferred ones.

In this case it would be:
```suggestion
  const [specieValue, setSpecieValue] = useState("");
```